encord/objects/ontology_labels_impl.py

```diff
@@ -0,0 +1,2906 @@
+
+        @property
+        def last_edited_by(self) -> Optional[str]:
+            self._check_if_frame_view_valid()
+            return self._get_object_frame_instance_data().object_frame_instance_info.last_edited_by
+
+        @last_edited_by.setter
+        def last_edited_by(self, last_edited_by: Optional[str]) -> None:
+            """
+            Set the last_edited_by field with a user email or None if it should default to the current user of the SDK.
+            """
+            self._check_if_frame_view_valid()
+            if last_edited_by is not None:
+                check_email(last_edited_by)
+            self._get_object_frame_instance_data().object_frame_instance_info.last_edited_by = last_edited_by
+
+        @property
+        def confidence(self) -> float:
+            self._check_if_frame_view_valid()
+            return self._get_object_frame_instance_data().object_frame_instance_info.confidence
+
+        @confidence.setter
+        def confidence(self, confidence: float) -> None:
+            self._check_if_frame_view_valid()
+            self._get_object_frame_instance_data().object_frame_instance_info.confidence = confidence
+
+        @property
+        def manual_annotation(self) -> bool:
+            self._check_if_frame_view_valid()
+            return self._get_object_frame_instance_data().object_frame_instance_info.manual_annotation
+
+        @manual_annotation.setter
+        def manual_annotation(self, manual_annotation: bool) -> None:
+            self._check_if_frame_view_valid()
+            self._get_object_frame_instance_data().object_frame_instance_info.manual_annotation = manual_annotation
+
+        @property
+        def reviews(self) -> List[dict]:
+            """
+            A read only property about the reviews that happened for this object on this frame.
+            DENIS: TODO: probably want to type this out and possibly lazy load this.
+            """
+            self._check_if_frame_view_valid()
+            return self._get_object_frame_instance_data().object_frame_instance_info.reviews
+
+        @property
+        def is_deleted(self) -> bool:
+            """This property is only relevant for internal use."""
+            self._check_if_frame_view_valid()
+            return self._get_object_frame_instance_data().object_frame_instance_info.is_deleted
+
+        def _get_object_frame_instance_data(self) -> ObjectInstance.FrameData:
+            return self._object_instance._frames_to_instance_data[self._frame]
+
+        def _check_if_frame_view_valid(self) -> None:
+            if self._frame not in self._object_instance._frames_to_instance_data:
+                raise LabelRowError(
+                    "Trying to use an ObjectInstance.FrameView for an ObjectInstance that is not on the frame."
+                )
+
+    @dataclass
+    class FrameInfo:
+        created_at: datetime = datetime.now()
```

Comment:
defaulting these might be dangerous, if we / the customer forget to set them, they'll default to whenever the program runs

Reply:
Interesting. I think the whole point of defaulting this is really to make the creation of objects super simple, not needing to worry about all of that.

Reply:
Ah. If it's only for creating them - I think that's fine (or at least up to design preference, as long as we're consistent). I'm thinking what would happen when we use these to wrap data coming from the BE